src/quickfix.c

```diff
@@ -7822,7 +7914,19 @@ qf_set_properties(qf_info_T *qi, dict_T *what, int action, char_u *title)
     if ((di = dict_find(what, (char_u *)"title", -1)) != NULL)
 	retval = qf_setprop_title(qi, qf_idx, what, di);
     if ((di = dict_find(what, (char_u *)"items", -1)) != NULL)
-	retval = qf_setprop_items(qi, qf_idx, di, action);
+    {
+	int keep_idx = FALSE;
+	dictitem_T *keep_idx_di;
+
+	if ((keep_idx_di = dict_find(what, (char_u *)"keep_idx", -1)) != NULL)
+	{
+	    if (keep_idx_di->di_tv.v_type != VAR_NUMBER)
+		return FAIL;
```

Comment:
I'm not sure if it's OK to bail out out `qf_set_properties` here. The other code just sets `retval = FAIL` and then goes on to possible do stuff after this block of `if` statments.

Feedback appreciated!

Reply:
Yeah,  I think following what has been done for the other properties is correct and should be done here as well.

Reply:
Ok, updated. This got kind of messy, I had to duplicate the call to `qf_setprop_items`.

I'm going to mark this resolved, but do let me know if you can think of a less bad way of doing this.

Reply:
For the record, this is irrelevant now that we're doing a `'u'` action instead of a `keep_idx` option.